session/tidb.go

```diff
@@ -174,6 +174,7 @@ func runStmt(ctx context.Context, sctx sessionctx.Context, s sqlexec.Statement)
 		} else {
 			err = se.CommitTxn(ctx)
 		}
+		se.sessionVars.TxnCtx.Cleanup()
```

Comment:
How about moving this inside the `CommitTxn` and `RollbackTxn` ?  `runStmt` is not always executed. @lysu